<launch>
    <arg name="goal_angle" default="$(optenv ROBOT_ROTATION_TEST_GOAL_ANGLE 5.0)" />
    <arg name="tolerance" default="$(optenv ROBOT_ROTATION_TEST_TOLERANCE 0.1)" />
    <arg name="angular_speed_z" default="$(optenv ROBOT_ROTATION_TEST_ANGULAR_SPEED 0.2)"/>
    <arg name="rotated_angle_topic" default="/rotated_angle"/>
    <arg name="goal_angle_topic" default="/goal_angle"/>
    <arg name="amcl_pose_topic" default="/amcl_pose"/>
    <arg name="velocity_topic" default="/cmd_vel"/>

    <arg name="sim_timeout_seconds" default="$(optenv ROBOT_ROTATION_TEST_SIM_TIMEOUT_SECONDS 300)" />
    <arg name="model_name" default="$(optenv MODEL_NAME /)" />
    <arg name="start_time_seconds" default="$(optenv ROBOT_ROTATION_TEST_START_TIME_SECONDS 10)" />
    <arg name="environment" default="$(optenv ENVIRONMENT AWS)" />

    <include file="$(find mp_move_manager)/launch/rotate.launch">
        <arg name="velocity_topic" value="$(arg velocity_topic)"/>
        <arg name="amcl_pose_topic" value="$(arg amcl_pose_topic)"/>
        <arg name="rotated_angle_topic" value="$(arg rotated_angle_topic)"/>
        <arg name="angular_speed_z" value="$(arg angular_speed_z)"/>
    </include>

      <!-- Test node params -->   
    <node pkg="mp_slippage_and_motor_saturation_tests" type="rotation_test_node.py" name="rotation_test" output="screen" required="true">
        <param name="sim_timeout_seconds" type="int" value="$(arg sim_timeout_seconds)" />
        <param name="model_name" type="string" value="$(arg model_name)" />
        <param name="start_time_seconds" type="int" value="$(arg start_time_seconds)" /> <!-- waiting for robot is spawned and stabilized-->
        <param name="goal_angle" value="$(arg goal_angle)"/>
        <param name="tolerance" value="$(arg tolerance)"/>
        <param name="environment" type="string" value ="$(arg environment)"/>
        <param name="rotated_angle_topic" value="$(arg rotated_angle_topic)"/>
        <param name="goal_angle_topic" value="$(arg goal_angle_topic)"/>
    </node>
 </launch>